Python/symtable.c

```diff
@@ -196,9 +196,17 @@ ste_dealloc(PyObject *op)
 
 #define OFF(x) offsetof(PySTEntryObject, x)
 
+static PyObject *
+ste_get_name(PySTEntryObject *ste, void *context)
+{
+    if (ste->ste_type == AnnotationBlock) {
+        return Py_NewRef(&_Py_ID(__annotate__));
+    }
+    return Py_NewRef(ste->ste_name);
+}
```

Comment:
I was assuming that the ste_name of an annotations ste would be set to `__annotations__` so you wouldn’t need to do this. Is that not possible?

Reply:
Your previous comment suggested using the ste_name to store the name of the corresponding function instead. We can't do both.

Reply:
Can we set the ste_name to `funcname.__annotations__`?

Reply:
Ok, I'm not sure which option is best. I'll approve with whichever you choose.

Reply:
There are at least two options, the initial version of this PR (adding `ste_function_name`) and the one I pushed earlier this morning after your comment (overloading `ste_name`). The advantage of the `ste_function_name` version is that it's simpler: less code and `ste_name` always means the same thing. The advantage of the version that overloads `ste_name` is that we save a little on memory usage: one fewer pointer in the symbol table struct.

Personally I prefer the `ste_function_name` version since it's simpler even if it uses slightly more memory during compilation, but if you prefer the other version, I'm OK with that.

Reply:
I'm happy with the original version. Thank you for indulging my musings.